spring-restdocs-core/src/main/java/org/springframework/restdocs/payload/FieldTypeResolver.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.restdocs.payload;
+
+import org.springframework.http.MediaType;
+
+/**
+ * Public abstraction for external access to field type determination for xml and json
+ * payloads.
+ *
+ * @author Mathias Düsterhöft
+ */
```

Comment:
This is new public API so it should be annotated with `@since`. I think I'd like to sneak this into 2.0.x, so it should be `@since 2.0.3`.